### DISCUSSION TITLE
Manifest issue in the jar file

### DISCUSSION THREAD

**Discussion by mdziczkowski** [rabbitmq/rabbitmq-java-client#911] [General]
After downloading the client from the link given on the rabbitmq website, with points to: [Maven repository](http://repo1.maven.org/maven2/com/rabbitmq/amqp-client/5.16.0/amqp-client-5.16.0.jar) and then trying to launch it using OpenJDK-11, I had recieved a following error related to the manifest:

```
no main manifest attribute, in ./amqp-client-5.16.0.jar
```

After extracting the jar file and opening the manifest file in a code editor, I saw that the content was missformed and I haven't seen the required to make it work `Main-class` attribute.

Could someone fix that?

Thanks

**Comment by michaelklishin** [comment]
This is a library, not an executable. There is no main file in the manifest because, well, there really is no main file.

Perhaps you are confusing this client with [PerfTest](https://github.com/rabbitmq/rabbitmq-perf-test), which is an executable (although there's also bits that can be used as a library).

**Reply by mdziczkowski** [reply]
I would like to ask in this case, why isn't there any. It would make the implementation much easier if there were a main class and function.

In my opinion, there should exist client binaries (and scripts in case of some programming languages) and the "libraries" should be moved into the developer section, to don't cause confusion

**Reply by acogoluegnes** [reply]
IIUC you expected a CLI tool to interact with RabbitMQ. That's not what the Java client is. The programming language of such a tool would not matter anyway. The [RabbitMQ website](https://rabbitmq.com/devtools.html#rabbitmq-cli) lists some tools like this.

> In my opinion, there should exist client binaries (and scripts in case of some programming languages) and the "libraries" should be moved into the developer section, to don't cause confusion

What are you referring to? A page on the website?

**Reply by CalmBubbles** [reply]
For no reason I am mentioned in this thread

**Reply by jack-temko** [reply]
There was a bot posting about some porn site, it mass spammed a bunch of people.
